background/services/chain/utils/quai-transactions.ts

```diff
@@ -80,48 +82,109 @@ export const createSerializedQuaiTransaction = (
     status,
     type,
     value,
+    gasPrice,
   } = transaction
 
   const serializedChainId = chainId ? chainId.toString() : ""
+  switch (status) {
```

Comment:
you can create a shared tx object with shared tx fields between all statuses, and then just add specifically that you need in each status.